### ISSUE TITLE
interoperability with other reactive libraries

### ISSUE THREAD

**Issue by niklas-wortmann** [JWO719/rxjs-web#3]
I would really like to get feedback if any other reactive community would be interested in something like this. Technically we could add some kind of interoperability with callbag or xstreams. I'm just mildly concerned about additional maintenance effort, but if it is a highly requested feature, I'd consider adding it.

Therefore I would really appreciate some kind of voting on this issue.

**Comment by chrisguttandin** [comment]
Hi Jan-Niklas,

I just came across this issue. If I remember correctly we once talked about this already at some point but I can't remember the outcome. 🤷‍♂️

I built a very similar library called [subscribable-things](https://github.com/chrisguttandin/subscribable-things). The main difference is that it doesn't depend directly on RxJS or any other library for reactive programming.

What do you think about using subscribable-things as the common ground wherever it makes sense? rxjs-web would then be responsible for making the APIs more rxjs-y (if that's word) and more convenient to use from within an app that's using RxJS already.

**Comment by niklas-wortmann** [comment]
Hi Chris, I actually created this issue based on our discussions. Regarding using subscribable-things, I am not sure about yet, I see several reasons against it (from my perspective):
 - dependency to a third party library (always comes with risks)
 - makes bug researches harder and even more difficult to fix
 - increased bundle size

I like what you are doing with subscribable thing but I am not sure yet how to move forward with this,

**Comment by chrisguttandin** [comment]
Yes, these are all valid points. Maybe I can weaken some of them. :-)

The bundle size should not be such an issue. subscribable-things is build with tree-shaking in mind. When only using a single function of it only that function should end up in the bundle.

I've written a lot of tests for all the provided functions. I guess the amount of code written to test each function outweighs the size of the actual source code to implement that function. The tests are written in JS which is probably the reason why GitHub things it's a JS project. :-)
